libraries/Microsoft.Bot.Builder.Skills/SkillConversation.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma warning disable CA2208 // Instantiate argument exceptions correctly
+
+using System;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Skills
+{
+    /// <summary>
+    /// Manages encoding ConversationId and ServiceUrl into packaged string for skill's conversation Id.
+    /// </summary>
+    internal class SkillConversation
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SkillConversation"/> class.
+        /// </summary>
+        public SkillConversation()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SkillConversation"/> class.
+        /// </summary>
+        /// <param name="packedConversationId">packed skill conversationId to unpack.</param>
+        public SkillConversation(string packedConversationId)
+        {
+            var parts = JsonConvert.DeserializeObject<string[]>(Encoding.UTF8.GetString(Convert.FromBase64String(packedConversationId)));
+            ConversationId = parts[0];
+            ServiceUrl = parts[1];
+        }
+
+        public string ConversationId { get; set; }
+
+        public string ServiceUrl { get; set; }
+
+        /// <summary>
+        /// Get packed skill conversationId.
+        /// </summary>
+        /// <returns>packed conversationId.</returns>
+        public string GetSkillConversationId()
+        {
+            if (String.IsNullOrEmpty(ConversationId))
+            {
+                throw new ArgumentNullException(nameof(ConversationId));
+            }
+
+            if (String.IsNullOrEmpty(ServiceUrl))
+            {
+                throw new ArgumentNullException(nameof(ServiceUrl));
+            }
+
+            return Convert.ToBase64String(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(new[]
```

Comment:
>ToBase64String [](start = 27, length = 14)

Does this need to be Base64 encoded versus just URL Encoded?  The JS version of skills potentially needs to run in a browser and it would be nice if we could avoid having to find a polyfill of some sort.

Reply:
Let me see what I can do here, I'll address this one once I merge back into the FristPass branch.

Reply:
The only one who creates and unpacks these is the SkillHostAdapter, no need for cross language support.